src/AWSExceptions.jl

```diff
@@ -12,4 +14,48 @@ struct InvalidFileName <: Exception
 end
 show(io::IO, e::InvalidFileName) = println(io, e.message)
 
+
+struct AWSException <: Exception
+    code
+    message
+    info
+    cause
+end
+show(io::IO, e::AWSException) = println(io, string(e.code,
+                       e.message == "" ? "" : (" -- " * e.message), "\n",
```

Comment:
Just use a multi-line function
```suggestion
                       isempty(e.message) ? "" : (" -- " * e.message), "\n",
```